// react-tag-manager/api/api.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _debug = require('debug');

var _debug2 = _interopRequireDefault(_debug);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var warn = (0, _debug2.default)('gtm:api:warning');
var log = (0, _debug2.default)('gtm:api:log');

exports.default = new (function () {
  function _class2() {
    var _this = this;

    _classCallCheck(this, _class2);

    this.queue = [];
    this.dataLayer = {};

    this.init = function (dataLayerName) {
      _this.dataLayerName = dataLayerName;
    };

    this.loaded = function () {
      var gtm = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : null;

      _this.gtm = gtm || window[_this.dataLayerName];
    };

    this.setDataLayer = function (data) {
      _this.dataLayer = _extends({}, _this.dataLayer, data);
    };

    this.trigger = function (data) {
      if (_this.gtm) {
        var eventData = _extends({}, _this.dataLayer, data);

        _this.gtm.push(eventData);

        log('Event triggered', _extends({}, eventData));
      }
    };
  }

  return _class2;
}())();
